### ISSUE TITLE
Missing Version ranges in `conan info . --json`

### ISSUE THREAD

**Issue by climblinne** [conan-io/conan#4442]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

In Conan 1.12.0 now `conan info . --json` is available.
In the normal output, you see the version ranges resolved:

```
Version ranges solved
    Version range '~1.12.1-dev' required by 'conanfile.py' resolved to 'mylib/1.12.1-dev.1+master.g06485fb@demo/testing'
```

It would be very useful to get this information also inside the json file.

**Comment by jgsogo** [comment]
The objective of the `--json` argument is to get the results of the command execution (the graph of dependencies), those lines you are referring to are more a log than a result and, as a log, those lines are still printed when you use the `--json` argument.

Anyway, what will be the use case? what do you need that information in the JSON for?

**Comment by climblinne** [comment]
On my CI line (gerrit) I get the information, which libraries are are changed and need to be validated. So I rebuild the libraries and want afterwards run all the tests, which are dependent on this one.
When I do a change on `Foo/1.1.1`  I need to know, if project `ProjA` would need to be triggered (e.g. requires `Foo/[^1.1.0]`).

**Comment by jgsogo** [comment]
I'm sorry, but I think I'm not understanding it and I want to because all the feedback related to CI is really valuable.

So, after building/testing your new library `Foo/1.1.1` you run a `conan info` over all your projects to see if some of them will make use of this new library because in the requirement section they could have a version range that gets resolved to this new library, isn't it?

If it is this way, why do you need to know about the version range instead of the library actually resolved? If `Foo/1.1.1` is already in your server, Conan will resolve the version range to use that version and it will be shown in the `json` output

**Comment by jgsogo** [comment]
I'm closing this issue, I think that the `conan info` should be enough to handle this situation, Conan will resolve the version ranges and inform about the resolved version. It is important to do it using Conan as other nodes in the graph can narrow the available versions for the range.

Thanks.
